Dockerfile

```diff
@@ -21,27 +15,40 @@ RUN git clone https://github.com/overleaf/overleaf.git \
 
 # Install dependencies needed to run configuration scripts
 # --------------------------------------------------------
-ADD ${baseDir}/package.json /var/www/package.json
-ADD ${baseDir}/git-revision.js /var/www/git-revision.js
-RUN cd /var/www && npm install
-
-
-# Replace overleaf/config/services.js with the list of available 
-# services in Overleaf Community Edition
-# --------------------------------------------------------------
-ADD ${baseDir}/services.js /var/www/sharelatex/config/services.js
-
-
+COPY root/build/ /
+RUN cd /var/www \
+&&    npm install \
+  \
 # Checkout services
 # -----------------
-RUN cd /var/www/sharelatex && \
-	npm install && grunt install;
-
-
+&&  cd /var/www/sharelatex \
+&&    npm install \
+&&    grunt install \
+  \
 # install and compile services
 # ----------------------------
-RUN bash -c 'cd /var/www/sharelatex && source ./bin/install-services'
-RUN bash -c 'cd /var/www/sharelatex && source ./bin/compile-services'
+&&  cd /var/www/sharelatex \
+&&    bash ./bin/install-services \
```

Comment:
checkout/install/compile are explicitly split for convenience. These are time consuming steps, and most of the maintenance work involves those stages, hen layer caching helps quite a lot. Unless it saves a significant amount of space they should remain separated.

Reply:
```
$ sudo docker run --rm --entrypoint sh sharelatex/sharelatex:2.1.1 -c 'find /var/www -name .git | xargs du -h -d 0' | sort -h
216K	/var/www/sharelatex/contacts/.git
260K	/var/www/sharelatex/notifications/.git
320K	/var/www/sharelatex/tags/.git
456K	/var/www/sharelatex/docstore/.git
536K	/var/www/sharelatex/real-time/.git
868K	/var/www/sharelatex/filestore/.git
900K	/var/www/sharelatex/track-changes/.git
964K	/var/www/sharelatex/chat/.git
996K	/var/www/sharelatex/.git
1.1M	/var/www/sharelatex/spelling/.git
1.2M	/var/www/sharelatex/document-updater/.git
2.7M	/var/www/sharelatex/clsi/.git
65M	/var/www/sharelatex/web/.git
```
Does 65MB qualify as 'significant amount of space'?

Reply:
Not really enough to sacrifice it in my opinion. However we can perhaps clean up git history just after checking out the repos?

Reply:
Yes.

Side note: We need to tweak the `git-revision` script in server-pro to amend the output file.